### DISCUSSION TITLE
Using app/router.options

### DISCUSSION THREAD

**Discussion by ijkml** [nuxt/nuxt#17270] [Questions] [answered]
Hello 

Following the guide at [Using App Router Options](https://v3.nuxtjs.org/guide/directory-structure/pages#using-approuteroptions), I have set up my `router.options.ts` as follows:

```ts
import type { RouterConfig } from '@nuxt/schema';

export default <RouterConfig>{
  scrollBehavior(to, _from, savedPosition) {
    if (to.hash) {
      return {
        el: to.hash,
        behavior: 'smooth',
        top: 64,
      };
    }
    if (savedPosition) {
      return savedPosition;
    }
    return { top: 0 };
  },
};
```

in the root dir, but it does not seem to work. I'm a little unsure, should the file be placed in the root dir or in a sub-folder named 'app'?

**Comment by konradsienkowski** [comment, marked as answer]
It should be placed in the `app` sub-folder, though it's not included in a default directory structure. It's described [here](https://v3.nuxtjs.org/guide/directory-structure/pages/#using-approuteroptions)

**Reply by ijkml** [reply]
Thank you for your reply! Yes, it works.

But now, I'm having problems with `savedPosition`. Using the back button, `savedPosition.top` is logged as a large number (eg. 4000) but the page just scrolls a few pixels from the top. I'm using the exact same function as above 👆. Is there anything I'm doing wrong?

**Reply by konradsienkowski** [reply]
@ijkml keeping the whole logic in one conditional should help: 
``` javascript

scrollBehavior(to, _from, savedPosition) {
  if (to.hash) {
    return {
      el: to.hash,
      behavior: 'smooth',
      top: 64,
    };
  } else if (savedPosition) {
    return savedPosition;
  } else {
    return { top: 0 };
  }
}
```
